### ISSUE TITLE
Popup Emoji above softkeyboard

### ISSUE THREAD

**Issue by tuhin10** [ankushsachdeva/emojicon#20]
Instead of showing just over the softkeyboard emoji popup is being shown just above the keyboard.
![screenshot_2015-05-11-01-47-37](https://cloud.githubusercontent.com/assets/11805182/7556080/3c5a56c8-f780-11e4-83f3-41b84720a743.png)

**Comment by kaushik-mota** [comment]
Same thing happened with me.

**Comment by surajnavkudkarpersonal** [comment]
I have same issue on some devices
![device-2015-07-16-163459](https://cloud.githubusercontent.com/assets/8064400/8721501/af0a83e6-2bd8-11e5-892f-7c8bfde3f1f5.png)

**Comment by cristiano939** [comment]
Same issue here!
But only when the activity is going to background

**Comment by remon** [comment]
did any one fix this problem ?

**Comment by Shruti6293** [comment]
any one resolved this  ?

**Comment by remon** [comment]
if anyone fixed it please tell me

**Comment by pdee2131** [comment]
Hi oh okay will do that ASAP

_Divine Pdee Asiedu_
_P.O.Box STC 73,_
_Accra-North,_
_00233273772879_
@pdee2131 pdee asiedu

**Comment by remon** [comment]
@pdee2131 ,I hope so

**Comment by remon** [comment]
@tuhin10  , @Kaushik-M , @suraj9781  ,did you fix it cause I am struggling with the same problem for 2 months now

**Comment by pdee2131** [comment]
Hi Reymond ,

I have still not figured it out so i just released the app, but it
currently supports android emoji's for now, which is compatible for devices
KITKAT and above since those devices already have their own emoji's
keyboard....You can contact me if you still need any help

_Divine Pdee Asiedu_
_P.O.Box STC 73,_
_Accra-North,_
_00233273772879_
@pdee2131 pdee asiedu

**Comment by mansoorahmadsamar** [comment]
I'm having the same problem, if anyone figure it out please let me know

**Comment by pdee2131** [comment]
Yhh the problem is still there for me too but i have seen that Whatsapp
implemented it and it works so am still trying to figure it out

_Divine Pdee Asiedu_
_P.O.Box STC 73,_
_Accra-North,_
_00233273772879_
@pdee2131 pdee asiedu

**Comment by VishalJogiya** [comment]
I've corrected above error atleast it worked for wherever I've tested. Here is link of my blog where I've written the corrections in this library. 
http://vjscrazzy.blogspot.in/2015/09/use-emoticon-emoji-library-in-android.html?m=1

**Comment by VishalJogiya** [comment]
5> Use EmojiconEdittext insted of EditText and EmojiconTextView in place of TextView, thats it. And a button to show emoji popup.

6> changes that have to made at this lib are ,MainActiviy.java

   emojiButton.setOnClickListener(new View.OnClickListener() {

```
        @Override
        public void onClick(View v) {

            //If popup is not showing => emoji keyboard is not visible, we need to show it
```

//
                //IF POPUP IS SHOWING, SIMPLY DISMISS IT TO SHOW THE UNDELYING TEXT KEYBOARD
                if (popup.isShowing()) {
                    popup.dismiss();
                    final InputMethodManager inputMethodManager = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
                    inputMethodManager.showSoftInputFromInputMethod(emojiconEditText.getWindowToken(), InputMethodManager.SHOW_IMPLICIT);

```
            } else {

                final InputMethodManager inputMethodManager = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
                inputMethodManager.hideSoftInputFromWindow(emojiconEditText.getWindowToken(), InputMethodManager.HIDE_IMPLICIT_ONLY);

                changeEmojiKeyboardIcon(emojiButton, R.drawable.ic_action_keyboard);
                popup.showAtBottom();
```

//                    emojiconEditText.setFocusableInTouchMode(true);
//                    emojiconEditText.requestFocus();

```
            }
        }
    });
```

  b> Add This code into onCreate method of mainactivity.java

   popup.setOutsideTouchable(true);
        popup.setFocusable(true);
        popup.getContentView().setFocusableInTouchMode(true);
        popup.getContentView().setOnKeyListener(new View.OnKeyListener() {
            @Override
            public boolean onKey(View v, int keyCode, KeyEvent event) {
                if (keyCode == KeyEvent.KEYCODE_BACK) {
                    popup.dismiss();
                    return true;
                }
                return false;
            }
        });

  c> delete  onSoftKeyboardListener method implementation from                           EmojiPopup in mainactivity.java.

   d> Inside EmojiconPopup.java constructor change

   I>setSoftInputMode(LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN);
   II>public void showAtBottom(){
    showAtLocation(rootView, Gravity.BOTTOM, 0, -1);
    }

**Comment by ankush19902** [comment]
i dont think it worked as the editext gets hidden. It position itself at the bottom and pop cover that view. So its just pop up which align itself to bottom. @VishalJogiya

**Comment by pdee2131** [comment]
Woow that's soo great to hear will try and implement it soon

_Divine Pdee Asiedu_
_P.O.Box STC 73,_
_Accra-North,_
_00233273772879_
@pdee2131 pdee asiedu

**Comment by VishalJogiya** [comment]
It works for me, you just try this solution  if you get any error then tell me. 
One more thing is that it only works for vertical orientation not for horizontal,somehow  you just need to make its width to match parent. 
Here I'm checking if the keyboard is open or not, 
If it does then forcefully close the keyboard and show emoji keyboard,. 
 if user touchs outside the emoji keyboard then closed it.
Same as what's app does. @ankush19902

**Comment by gitbiraz** [comment]
its not working. please provide another solution

**Comment by ductranit** [comment]
This issue is fixed on my side by change 
`setSoftInputMode(LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);` 
to 
  `setSoftInputMode(LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN)`
in  [EmojiconsPopup](https://github.com/ankushsachdeva/emojicon/blob/master/lib/src/github/ankushsachdeva/emojicon/EmojiconsPopup.java#L81)
